doc/source/whatsnew/v1.2.0.rst

```diff
@@ -613,6 +613,7 @@ Indexing
 - Bug in indexing on a :class:`Series` or :class:`DataFrame` with a :class:`CategoricalIndex` using listlike indexer that contains elements that are in the index's ``categories`` but not in the index itself failing to raise ``KeyError`` (:issue:`37901`)
 - Bug in :meth:`DataFrame.iloc` and :meth:`Series.iloc` aligning objects in ``__setitem__`` (:issue:`22046`)
 - Bug in :meth:`DataFrame.loc` did not raise ``KeyError`` when missing combination was given with ``slice(None)`` for remaining levels (:issue:`19556`)
+- Bug in :meth:`DataFrame.loc` and :meth:`DataFrame.__getitem__` returning wrong elements for non-monotonic :class:`DatetimeIndex` for string slices (:issue:`33146`)
```

Comment:
is Series not affected?

Reply:
"wrong elements" -> "incorrect elements" 

"instead of raising FooError"?

Reply:
No this should work, not raising. Yes Series is affected too.